python/arcticdb/storage_fixtures/s3.py

```diff
@@ -0,0 +1,275 @@
+        b.slow_cleanup(failure_consequence="We will be charged unless we manually delete it. ")
+
+
+class MotoS3StorageFixtureFactory(BaseS3StorageFixtureFactory):
+    default_key = Key("awd", "awd", "dummy")
+    _RO_POLICY: str
+    _RW_POLICY: str
+    host = "localhost"
+    region = "us-east-1"
+    port: int
+    endpoint: str
+    _enforcing_permissions = False
+    _iam_endpoint: str
+    _p: multiprocessing.Process
+    _s3_admin: Any
+    _iam_admin: Any = None
+    _bucket_id = 0
+    _live_buckets: List[S3Bucket] = []
+
+    @staticmethod
+    def run_server(port):
+        import werkzeug
+        from moto.server import DomainDispatcherApplication, create_backend_app
+
+        class _HostDispatcherApplication(DomainDispatcherApplication):
+            """The stand-alone server needs a way to distinguish between S3 and IAM. We use the host for that"""
+
+            _MAP = {"localhost": "s3", "127.0.0.1": "iam"}
+
+            _reqs_till_rate_limit = -1
+
+            def get_backend_for_host(self, host):
+                return self._MAP.get(host, host)
+
+            def __call__(self, environ, start_response):
+                path_info: bytes = environ.get("PATH_INFO", "")
+
+                with self.lock:
+                    if path_info in ("/rate_limit", b"/rate_limit"):
+                        length = int(environ["CONTENT_LENGTH"])
+                        body = environ["wsgi.input"].read(length).decode("ascii")
+                        self._reqs_till_rate_limit = int(body)
+                        start_response("200 OK", [("Content-Type", "text/plain")])
+                        return [b"Limit accepted"]
+
+                    if self._reqs_till_rate_limit == 0:
+                        start_response("503 Slow down", [("Content-Type", "text/plain")])
+                        return [b"Simulating rate limit"]
+                    else:
+                        self._reqs_till_rate_limit -= 1
+
+                return super().__call__(environ, start_response)
+
+        werkzeug.run_simple(
+            "0.0.0.0",
+            port,
+            _HostDispatcherApplication(create_backend_app),
+            threaded=True,
+            ssl_context=None,
+        )
+
+    def _safe_enter(self):
+        port = self.port = get_ephemeral_port()
+        self.endpoint = f"http://{self.host}:{port}"
+        self._iam_endpoint = f"http://127.0.0.1:{port}"
+
+        p = self._p = multiprocessing.Process(target=self.run_server, args=(port,))
+        p.start()
+        wait_for_server_to_come_up(self.endpoint, "moto", p)
+
+        self._s3_admin = self._boto("s3", self.default_key)
+        return self
+
+    def __exit__(self, exc_type, exc_value, traceback):
+        GracefulProcessUtils.terminate(self._p)
+
+    def _create_user_get_key(self, user: str, iam=None):
+        iam = iam or self._iam_admin
+        user_id = iam.create_user(UserName=user)["User"]["UserId"]
```

Comment:
`user_id` is unused, do you mean to include it in the return value?

Reply:
Useful for debug